demo/components/checkbox-demo.tsx

```diff
@@ -3,50 +3,156 @@ import {observable} from 'mobx'
 import {observer} from 'mobx-react'
 import {CheckBox, CheckBoxIconProps} from "../../src";
 
-export const demoCheckBoxText: string = 'I agree to the terms above';
-export const demoCheckBoxInitialValue: boolean = false;
+export const demoCheckBoxText: string = "Yes, I'm over 18 years old";
 
 const style = require('./checkbox-demo.st.css').default;
 
 @observer
-export class CheckBoxDemo extends React.Component<{},{value: boolean}> {
+export class CheckBoxDemo extends React.Component<{},{}> {
+    render() {
+        return (
+            <div>
+                <div>
+                    <h3>Basic CheckBox</h3>
+                    <BasicDemo/>
+                </div>
+
+                <div>
+                    <h3>Disabled</h3>
+                    <DisabledDemo/>
+                </div>
+
+                <div>
+                    <h3>Indeterminate</h3>
+                    <IndeterminateDemo/>
+                </div>
+
+                <div>
+                    <h3>Custom Icons</h3>
+                    <CustomIconsDemo/>
+                </div>
+            </div>
+        )
+    }
+}
 
-    @observable value: boolean = demoCheckBoxInitialValue;
+@observer
+class BasicDemo extends React.Component<{}, {}> {
+    @observable value: boolean = false;
 
-    handleChange = (val: boolean) => {this.value = val};
+    handleChange = (val: boolean) => { this.value = val };
 
     render() {
         return (
-            <div>
+            <div data-automation-id="BASIC_DEMO">
                 <CheckBox value={this.value}
-                          text={demoCheckBoxText}
-                          boxIcon={CheckBoxSVG}
-                          tickIcon={TickMarkSVG}
-                          onChange={this.handleChange}/> <br/>
+                          onChange={this.handleChange}>
+                    <span data-automation-id="BASIC_LABEL" className={style.label}>{demoCheckBoxText}</span>
+                </CheckBox> <br/>
                 <button disabled={!this.value} data-automation-id="BUTTON_SUBMIT">
-                    Submit
+                    Proceed
                 </button>
             </div>
         )
     }
 }
 
-const CheckBoxSVG: React.SFC<CheckBoxIconProps> = (props) => {
+@observer
+class DisabledDemo extends React.Component<{}, {}> {
+    @observable value: boolean = false;
+
+    handleChange = (val: boolean) => { this.value = val };
+
+    render() {
+        return (
+            <div>
+                <span data-automation-id="DISABLED_DEMO">
+                    <CheckBox value={this.value}
```

Comment:
that was my point... setting it to constant value is cheating 😄 .  The component test makes sure that clicking it doesn't cause the tick mark to appear